### PR TITLE
Allow a user to change their email address when logged in

### PR DESCRIPTION
**Pull request by chrisveilleux**
## Description
New feature that allows a user to change their email address in "My Mycroft".  

## How to test
Change your email address in ""My Mycroft" (must be done in tandem with change to selene-backend repository)

### CODE REVIEW

File: projects/account/src/app/core/http/profile.service.ts [hunk 1]
```diff
@@ -87,4 +89,18 @@ export class ProfileService {
...
+        return this.http.get(VALIDATE_EMAIL_URL, {params: queryParams});
+    }
+
+    changeEmailAddress(newEmailAddress: string) {
+        const codedEmailAddress = btoa(newEmailAddress);
+        return this.http.put(CHANGE_EMAIL_ADDRESS_URL, {token: codedEmailAddress});
+    }
+
+    verifyEmailAddress(newEmailAddress: string) {
+        return this.http.put<string>(VERIFY_EMAIL_URL, {token: newEmailAddress});
```
Comment (krisgesling) [inline projects/account/src/app/core/http/profile.service.ts:104]: looks like a typo with `<string>`

File: projects/account/src/app/modules/profile/components/cards/login/login.component.ts [hunk 2]
```diff
@@ -61,24 +59,37 @@ export class LoginComponent {
...
+                if (newEmailAddress) {
+                    this.updateEmailAddress(newEmailAddress);
+                }}
+        );
+    }
+
+    updateEmailAddress (newEmailAddress): void {
+        this.profileService.changeEmailAddress(newEmailAddress).subscribe({
+            next: () => { this.openSnackbar('info', 'Check your inbox for a link to verify the new email address'); },
+            error: () => { this.openSnackbar('error', 'An error occurred changing the email address'); }
```
Comment (krisgesling) [inline projects/account/src/app/modules/profile/components/cards/login/login.component.ts:85]: Any idea what these errors might be, or is this a bit of a catch all?

File: projects/account/src/app/modules/profile/pages/verify-email/verify-email.component.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    ngOnInit(): void {
+        const resetToken = this.route.snapshot.queryParams['token'];
+        this.profileService.verifyEmailAddress(resetToken).subscribe({
+            next: () => {
+                this.verificationMessage = 'Thank you for verifying your new email address.  Use it the next time you login.';
+                this.emailVerified = true;
+            },
+            error: (error) => {
+                if (error.status === 401) {
+                    this.verificationMessage = 'Your need to be logged in to verify your email address.  ' +
```
Comment (krisgesling) [inline projects/account/src/app/modules/profile/pages/verify-email/verify-email.component.ts:33]: "Your" > "You"

File: projects/account/src/app/modules/profile/pages/verify-email/verify-email.component.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        private route: ActivatedRoute,
+        private profileService: ProfileService,
+        private router: Router
+    ) { }
+
+    ngOnInit(): void {
+        const resetToken = this.route.snapshot.queryParams['token'];
+        this.profileService.verifyEmailAddress(resetToken).subscribe({
+            next: () => {
+                this.verificationMessage = 'Thank you for verifying your new email address.  Use it the next time you login.';
```
Comment (krisgesling) [inline projects/account/src/app/modules/profile/pages/verify-email/verify-email.component.ts:28]: Are we verifying the address on account creation too? 
If so might remove the word "new".

**Review by krisgesling — COMMENTED** [review]
Looks great - only a couple of tiny things

File: projects/account/src/app/core/http/profile.service.ts (hunk 1 above)
Comment (chrisveilleux) [reply projects/account/src/app/core/http/profile.service.ts:104]: not a typo, it is how you indicate that an observable string object will be returned

File: projects/account/src/app/modules/profile/components/cards/login/login.component.ts (hunk 2 above)
Comment (chrisveilleux) [reply projects/account/src/app/modules/profile/components/cards/login/login.component.ts:85]: Don't know of anything specific that might go wrong.  It's a catch all until more specific errors are identified and can be added.

File: projects/account/src/app/modules/profile/pages/verify-email/verify-email.component.ts (hunk 4 above)
Comment (chrisveilleux) [reply projects/account/src/app/modules/profile/pages/verify-email/verify-email.component.ts:28]: This is just for the email verification step when changing the email address.

**Pull request merged**